llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -1943,6 +1943,30 @@ static bool swapICmpOperandsToExposeCSEOpportunities(CmpInst *Cmp) {
   return false;
 }
 
+static bool foldFCmpToFPClassTest(CmpInst *Cmp) {
+  FCmpInst *FCmp = dyn_cast<FCmpInst>(Cmp);
+  if (!FCmp)
+    return false;
+
+  // Reverse the canonicalization if it is a FP class test
```

Comment:
This should consider the target's operation legality

Reply:
This probably should avoid this is fabs is free and the compare is legal

Reply:
This change wasn't made?

Reply:
Done.